[macOS] Added CodeQL Action Bundle to macOS 11 and macOS 12

erik-bershel opened the pull request:
# Description
Added CodeQL Action bundle to cached tools. Only partly tested.

#### Related issue:
Previous PR issue #6059

## Virtual environments affected
- [ ] macOS 10.15
- [x] macOS 11
- [x] macOS 12

## Check list
- [x] Related issue / work item is attached
- [x] Tests are written (if applicable)
- [x] Documentation is updated (if applicable)
- [ ] Changes are tested and related VM images are successfully generated

al-cheb commented on images/macos/toolsets/toolset-11.json:
```diff
@@ -296,6 +296,14 @@
             ]
+        },
+        {
+            "name": "CodeQL",
```
Where do we use it?

erik-bershel replied:
Removed this part as recommended.

al-cheb commented on images/macos/software-report/SoftwareReport.Toolcache.psm1:
```diff
@@ -57,6 +65,8 @@ function Build-ToolcacheSection {
     $output += New-MDList -Lines (Get-ToolcacheNodeVersions) -Style Unordered
     $output += New-MDHeader "Go" -Level 4
     $output += Get-ToolcacheGoTable
+    $output += New-MDHeader "CodeQL" -Level 4  
```
Let's for consistency use the output like for Windows and Linux.

al-cheb commented on images/macos/provision/core/codeql-bundle:
```diff
@@ -0,0 +1,23 @@
+#!/bin/bash -e -o pipefail
```
Where do you update packer templates to add this script?

al-cheb replied:
Let's for consistency add the .sh extension

al-cheb commented on images/macos/toolsets/toolset-11.json:
```diff
@@ -296,6 +296,14 @@
+            "arch": "x64",
+            "platform" : "darwin",
+            "versions": [
+                "*"
```
How do we plan to install multiple versions?

al-cheb commented on images/macos/software-report/SoftwareReport.Toolcache.psm1:
```diff
@@ -44,6 +44,14 @@ function Get-ToolcacheGoTable {
     return $Content
 }
 
+function Get-CodeQLBundleVersion {
```
If we have an array of versions why do we return only one version?

al-cheb commented on images/macos/toolsets/toolset-12.json:
```diff
@@ -241,6 +241,14 @@
+        {
+            "name": "CodeQL",
+            "arch": "x64",
+            "platform" : "darwin",
```
Where do we use this field?

al-cheb commented on images/macos/software-report/SoftwareReport.Toolcache.psm1:
```diff
@@ -44,6 +44,14 @@ function Get-ToolcacheGoTable {
+    $CodeQLVersionPath = Get-ChildItem $CodeQLVersionsWildcard | Select-Object -First 1 -Expand FullName
+    $CodeQLPath = Join-Path $CodeQLVersionPath -ChildPath "x64" | Join-Path -ChildPath "codeql" | Join-Path -ChildPath "codeql"
+    $CodeQLVersion = & $CodeQLPath version --quiet
+    return "CodeQL Action Bundle $CodeQLVersion"
```
If we are planning to use multiple version it should be like:

CodeQL Action Bundle:
- 0.0.0.1
- 0.0.0.2
- etc

erik-bershel commented:
@al-cheb changes:
1) Added extension to shell-script file codeql-bundle
2) Rewrote Software Reporter section
3) Rewrote test to simplier version without versions control
4) Added changes to provision scripts

mikhailkoliada commented on images/macos/provision/core/codeql-bundle.sh:
```diff
@@ -0,0 +1,23 @@
+codeqlToolcachePath="$AGENT_TOOLSDIRECTORY/codeql/$codeql_bundle_version/x64"
+mkdir -p $codeqlToolcachePath
+
+echo "Unzip CodeQL archive..."
```
forthcoming command is not unzip

erik-bershel replied:
Corrected the wording to be closer to the truth.

ddobranic commented:
/azp run macos11. macos12

ddobranic commented:
/azp run macos11, macos12

ddobranic commented:
/azp run macos11, macos12
